File: src/22_fibonacci_tree/fibonacci_tree.py
def fibonacci(n: int):
    temp = 0
    a = 0
    b = 1

    for _ in range(1, n):
        temp = a + b
        a = b
        b = temp

    return b

def main():
    n = int(input())
    result = 0
    for i in range(1, n):
        result += fibonacci(i)
    
    print(result)

if __name__ == "__main__":
    main()

File: src/22_fibonacci_tree/fibonacci_tree.cpp
#include <iostream>

using namespace std;

int fibonacci(int n) {
    long int temp = 0;
    long int a = 0;
    long int b = 1;

    for (int i = 1; i < n; ++i) {
        temp = a + b;
        a = b;
        b = temp;
    }

    return b;
}

int main(void) {
    int n = 0;
    cin >> n;
    long int result = 0;
    for (int i = 1; i < n; ++i) {
        result += fibonacci(i);
    }
    cout << result << endl;
    return 0;
}